src/lib/free-queue/test/free-queue.test.js

```diff
@@ -0,0 +1,162 @@
+import { expect } from 'https://cdnjs.cloudflare.com/ajax/libs/chai/5.1.1/chai.js';
+import { FreeQueue, MAX_CHANNEL_COUNT, RENDER_QUANTUM_FRAMES } from '../free-queue.js';
+
+// Mock WASM module
+const mockWasmModule = {
+  // Simulate memory allocation
+  _malloc: (size) => new ArrayBuffer(size), 
+  // Simulate memory deallocation
+  _free: () => {}, 
+  // Simulate HEAPF32
+  HEAPF32: new Float32Array(1024), 
+};
+
+describe('FreeQueue Class', () => {
+  const bufferLength = 512;
+  const channelCount = 2;
+  const maxChannelCount = 4;
+  let freeQueue = null;
+
+  beforeEach(() => {
+    freeQueue = new FreeQueue(mockWasmModule, bufferLength, channelCount, maxChannelCount);
+  });
+
+  afterEach(() => {
+    freeQueue.free();
+  });
+
+  describe('Initialization', () => {
+    it('should initialize with correct properties', () => {
+      expect(freeQueue.length).to.equal(bufferLength);
+      expect(freeQueue.numberOfChannels).to.equal(channelCount);
+      expect(freeQueue.maxChannelCount).to.equal(maxChannelCount);
+    });
+
+    it('should allocate the correct amount of memory', () => {
+      const dataByteSize = channelCount * bufferLength * Float32Array.BYTES_PER_ELEMENT;
+      expect(freeQueue.getPointer()).to.be.instanceof(ArrayBuffer);
+      expect(freeQueue.getPointer().byteLength).to.equal(dataByteSize);
+    });
+  });
+
+  describe('Channel Adaptation', () => {
+    it('should adapt to a new channel count within limits', () => {
+      freeQueue.adaptChannel(3);
+      expect(freeQueue.numberOfChannels).to.equal(3);
+    });
+  });
+
+  describe('Pushing and Pulling Data', () => {
+    it('should correctly push and pull data', () => {
```

Comment:
Let's match these tests with the 'push' and 'pull' sections from the FreeQueueSAB suite.

Reply:
Should we just define 2 sperate describe blocks for push and pull with the same test cases?

Reply:
Yes, I think when the APIs for the classes are the same we should use the same tests and format.  So since we did this in two blocks for SAB, I think we should do it in two blocks here too.

Reply:
Ok, I'll update this!